### DISCUSSION TITLE
Turkish language support

### DISCUSSION THREAD

**Discussion by dreixin** [emsesp/EMS-ESP32#868] [Ideas]
I am trying to add Turkish language support to EMS-ESP. Anyone willing to help is welcome.

**Comment by proddy** [comment]
@MichaelDvP do you think we have enough Flash space left for another language?

**Reply by MichaelDvP** [reply]
With the latest fixes i have 32k flash left, i think a new language is around 10-15k.

**Reply by proddy** [reply]
thanks, @dreixin see https://emsesp.github.io/docs/Coding/#adding-a-new-language and reach out on this discussion if you need help. thanks!

**Comment by bbqkees** [comment]
What would be the best way forward if we have more languages than memory space?
F.i. I sold only a few products to Turkey and Norway, but many more to like Hungary, Czech Republic etc.
At some point these will probably get translated as well.

The latest S32 has 16MB memory, but all others existing Gateways have 4MB.

**Reply by MichaelDvP** [reply]
For the 4MB versions we can use a smaller factory partition for loader and and a larger OTA-partion for ems-esp.
Actually this gives 700kB etxtra to ems-esp.
https://github.com/MichaelDvP/EMS-ESP-Loader
Update procedure is then 2 step: 
- system reboot -> boot loader
![grafik](https://user-images.githubusercontent.com/59284019/210211046-5deff791-5b24-4cd3-9c44-a306c8b03d8b.png)
- after reboot upload new file.

**Reply by proddy** [reply]
referencing https://github.com/emsesp/EMS-ESP32/issues/608
